ingest/src/main/java/com/microsoft/azure/kusto/ingest/StreamingIngestClient.java

```diff
@@ -132,13 +132,13 @@ public IngestionResult ingestFromStream(StreamSourceInfo streamSourceInfo, Inges
                     !(streamSourceInfo.getCompressionType() == null || !streamSourceInfo.isLeaveOpen()));
 
         } catch (DataClientException | IOException e) {
-            log.error(e.getMessage(), e);
-            throw new IngestionClientException(e.getMessage(), e);
-        } catch (DataServiceException e) {
             log.error(e.getMessage(), e);
             if (e.getCause() instanceof DataWebException && "Error in post request".equals(e.getMessage())) {
                 validateEndpointServiceType(connectionDataSource, EXPECTED_SERVICE_TYPE);
             }
+            throw new IngestionClientException(e.getMessage(), e);
```

Comment:
maybe in another PR
IngestionClientException should also have isPermanent and could actually just pass the ispermanent of its exception

Reply:
Agreed. As discussed, one of us will do this in another PR. I'm making a note of it.